src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/WrapGh.hpp

```diff
@@ -0,0 +1,191 @@
+      tmpl::list<TimeDerivSpatialMetric> /*meta*/,
+      const IntermediateVars& intermediate_vars) const noexcept;
+
+  tuples::TaggedTuple<DerivSpatialMetric> variables(
+      const tnsr::I<DataVector, volume_dim>& /*x*/, double /*t*/,
+      tmpl::list<DerivSpatialMetric> /*meta*/,
+      const IntermediateVars& intermediate_vars) const noexcept;
+
+  tuples::TaggedTuple<
+      gr::Tags::InverseSpatialMetric<volume_dim, Frame::Inertial, DataVector>>
+  variables(const tnsr::I<DataVector, volume_dim>& /*x*/, double /*t*/,
+            tmpl::list<gr::Tags::InverseSpatialMetric<
+                volume_dim, Frame::Inertial, DataVector>> /*meta*/,
+            const IntermediateVars& intermediate_vars) const noexcept;
+
+  tuples::TaggedTuple<
+      gr::Tags::ExtrinsicCurvature<volume_dim, Frame::Inertial, DataVector>>
+  variables(const tnsr::I<DataVector, volume_dim>& /*x*/, double /*t*/,
+            tmpl::list<gr::Tags::ExtrinsicCurvature<volume_dim, Frame::Inertial,
+                                                    DataVector>> /*meta*/,
+            const IntermediateVars& intermediate_vars) const noexcept;
+
+  tuples::TaggedTuple<gr::Tags::SqrtDetSpatialMetric<DataVector>> variables(
+      const tnsr::I<DataVector, volume_dim>& /*x*/, double /*t*/,
+      tmpl::list<gr::Tags::SqrtDetSpatialMetric<DataVector>> /*meta*/,
+      const IntermediateVars& intermediate_vars) const noexcept;
+
+  tuples::TaggedTuple<
+      gr::Tags::SpacetimeMetric<volume_dim, Frame::Inertial, DataVector>>
+  variables(const tnsr::I<DataVector, volume_dim>& /*x*/, double /*t*/,
+            tmpl::list<gr::Tags::SpacetimeMetric<volume_dim, Frame::Inertial,
+                                                 DataVector>> /*meta*/,
+            const IntermediateVars& intermediate_vars) const noexcept;
+
+  tuples::TaggedTuple<
+      GeneralizedHarmonic::Tags::Pi<volume_dim, Frame::Inertial>>
+  variables(const tnsr::I<DataVector, volume_dim>& /*x*/, double /*t*/,
+            tmpl::list<GeneralizedHarmonic::Tags::Pi<volume_dim,
+                                                     Frame::Inertial>> /*meta*/,
+            const IntermediateVars& intermediate_vars) const noexcept;
+
+  tuples::TaggedTuple<
+      GeneralizedHarmonic::Tags::Phi<volume_dim, Frame::Inertial>>
+  variables(
+      const tnsr::I<DataVector, volume_dim>& /*x*/, double /*t*/,
+      tmpl::list<
+          GeneralizedHarmonic::Tags::Phi<volume_dim, Frame::Inertial>> /*meta*/,
+      const IntermediateVars& intermediate_vars) const noexcept;
+
+ public:
+  // clang-tidy: google-runtime-references
+  void pup(PUP::er& p) noexcept { SolutionType::pup(p); }  // NOLINT
+};
+
+template <typename SolutionType>
+inline constexpr bool operator==(const WrapGh<SolutionType>& lhs,
+                                 const WrapGh<SolutionType>& rhs) noexcept {
+  return static_cast<SolutionType>(lhs) == static_cast<SolutionType>(rhs);
```

Comment:
`static_cast`s -> `dynamic_cast<const SolutionType&>` since this is runtime polymorphism and we only need to interpret the pointer differently.